ruma-events/src/room/aliases.rs

```diff
@@ -3,16 +3,113 @@
 use ruma_events_macros::StateEventContent;
 use ruma_identifiers::RoomAliasId;
 use serde::{Deserialize, Serialize};
+use serde_json::value::RawValue as RawJsonValue;
 
-use crate::StateEvent;
+use crate::{
+    EventContent, RedactedEventContent, RedactedRoomEventContent, RedactedStateEventContent,
+    StateEvent,
+};
 
 /// Informs the room about what room aliases it has been given.
 pub type AliasesEvent = StateEvent<AliasesEventContent>;
 
 /// The payload for `AliasesEvent`.
 #[derive(Clone, Debug, Deserialize, Serialize, StateEventContent)]
 #[ruma_event(type = "m.room.aliases")]
+#[ruma_event(custom_redacted)]
 pub struct AliasesEventContent {
     /// A list of room aliases.
     pub aliases: Vec<RoomAliasId>,
 }
+
+/// An aliases event that has been redacted.
+#[derive(Clone, Debug, Serialize)]
+pub struct RedactedAliasesEventContent;
+
+impl EventContent for RedactedAliasesEventContent {
+    fn event_type(&self) -> &str {
+        "m.room.aliases"
+    }
+
+    fn from_parts(
+        event_type: &str,
+        _content: Box<RawJsonValue>,
+    ) -> Result<Self, serde_json::Error> {
+        if event_type != "m.room.aliases" {
+            return Err(::serde::de::Error::custom(format!(
+                "expected event type `m.room.aliases`, found `{}`",
+                event_type
+            )));
+        }
+
+        Ok(Self)
+    }
+}
+
+impl RedactedEventContent for RedactedAliasesEventContent {
+    fn has_serialize_fields(&self) -> bool {
+        false
+    }
+
+    fn has_deserialize_fields() -> bool {
+        false
+    }
+
+    fn redacted(event_type: &str) -> Result<Self, serde_json::Error> {
+        if event_type != "m.room.aliases" {
+            return Err(::serde::de::Error::custom(format!(
+                "expected event type `m.room.aliases`, found `{}`",
+                event_type
+            )));
+        }
+
+        Ok(Self)
+    }
+}
+
+impl RedactedRoomEventContent for RedactedAliasesEventContent {}
+
+impl RedactedStateEventContent for RedactedAliasesEventContent {}
+
+/// An aliases event that has been redacted.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct RedactedV1AliasesEventContent {
```

Comment:
So you have added this second redacted aliases event content struct, but it's not actually part of `AnyRedactedStateEvent`, is it?

Reply:
Correct that's pretty pointless and should probably be it's own PR? Good to know we have all the plumbing to make it work though.

Reply:
I'm still sceptical about that working. Could you implement `RedactedAliasesEventContent` as containing an `Option<Vec<RoomAliasId>>` in this PR, and then move this to its own PR?

Reply:
Sounds good! Add the `Option<...>` then just leave it for later.